specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json

```diff
@@ -0,0 +1,212 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Trunk health Service",
+    "description": "Trunk health Service",
+    "contact": {
+      "email": "acsdevexdisc@microsoft.com"
+    },
+    "version": "2022-07-01-preview1"
+  },
+  "paths": {
+    "/sip/trunksHealth": {
+      "get": {
+        "tags": [
+          "TrunksHealth"
+        ],
+        "description": "This endpoint gets a list of all Trunks health configured for resource.",
+        "summary": "Gets a list of all Trunks health configured for resource.",
+        "operationId": "GetTrunksHealth",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Ok.",
+            "schema": {
+              "$ref": "#/definitions/ListOfTrunksHealth"
+            }
+          },
+          "default": {
+            "description": "Failure",
+            "schema": {
+              "$ref": "../../../Common/stable/2021-03-07/common.json#/definitions/CommunicationErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get a list of all Trunks health configured for resource.": {
+            "$ref": "./examples/GetTrunksHealths.json"
+          }
+        }
+      }
+    },
+    "/sip/trunksHealth/{fqdn}": {
+      "get": {
+        "tags": [
+          "TrunkHealth"
+        ],
+        "description": "This endpoint gets single Trunk health configured for resource.",
+        "summary": "Get single Trunk health configured for resource.",
+        "operationId": "GetTrunkHealth",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "path",
+            "name": "fqdn",
+            "description": "Case insensitive FQDN of the trunk.",
+            "required": true,
+            "type": "string",
+            "maxLength": 256
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Ok.",
+            "schema": {
+              "$ref": "#/definitions/TrunkHealth"
+            }
+          },
+          "default": {
+            "description": "Failure",
+            "schema": {
+              "$ref": "../../../Common/stable/2021-03-07/common.json#/definitions/CommunicationErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Gets Trunk health.": {
+            "$ref": "./examples/GetTrunksHealth.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ListOfTrunksHealth": {
+      "description": "Health of SIP trunks.",
+      "type": "object",
+      "properties": {
+        "value": {
```

Comment:
The guidelines are fairly clear that we should paginate all collections.

I do see below that we're stating the size of this collection is limited to 250. How hard is this limit? If it's set by laws of physics / a fairly stable standard, we don't need to paginate. If it's based more on current requirements, we should paginate.